shared/src/pool_fetching.rs

```diff
@@ -165,20 +166,38 @@ impl BaselineSolvable for Pool {
     }
 }
 
-type TokenPoolsMap = HashMap<TokenPair, Vec<Pool>>;
+const MAX_CACHED_BLOCKS: usize = 10;
 
-// Read though Pool Fetcher that keeps previously fetched pools in a cache, which get invalidated whenever there is a new block
+// Read though Pool Fetcher that keeps previously fetched pools in a LRU cache. Pools fetched for `BlockNumber::Latest` get invalidated whenever there is a new block
 pub struct CachedPoolFetcher {
     inner: Box<dyn PoolFetching>,
-    cache: Arc<Mutex<(Block, TokenPoolsMap)>>,
+    cache: Arc<Mutex<Cache>>,
     block_stream: CurrentBlockStream,
 }
 
+struct Cache {
+    /// Used to store details (e.g. hash) about the latest block. Needed so we know what `BlockNumber::Latest` refers to
+    latest_block: CurrentBlock,
+    pools: LruCache<u64, HashMap<TokenPair, Vec<Pool>>>,
```

Comment:
This might also be good to use in `BalancerPools` cache that is currently under construction.

Reply:
Cool, have we already discussed if each liquidity type should have its own cache or if there should be a cache for all `BaselineSolvable` types?